src/main/java/org/spongepowered/common/event/tracking/phase/plugin/PluginPhase.java

```diff
@@ -35,6 +35,7 @@
         public static final IPhaseState<BasicPluginContext> CUSTOM_SPAWN = new BasicPluginState();
         public static final IPhaseState<BasicPluginContext> SCHEDULED_TASK = new ScheduledTaskPhaseState();
         public static final IPhaseState<BasicPluginContext> TELEPORT = new BasicPluginState();
+        public static final IPhaseState<BasicPluginContext> FAKE_PLAYER = new BasicPluginState();
```

Comment:
It might be benefiting to have a custom state here that will push the fake player as a source and still ensure the plugin container as the context so to speak. You can kinda view this by overriding the popularCauseFrame method